staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter.go

```diff
@@ -304,8 +344,27 @@ func interfaceFromUnstructured(sv, dv reflect.Value) error {
 	return nil
 }
 
-func (c *Converter) ToUnstructured(obj runtime.Object, u *map[string]interface{}) error {
-	return toUnstructured(reflect.ValueOf(obj).Elem(), reflect.ValueOf(u).Elem())
+func (c *ConverterImpl) ToUnstructured(obj runtime.Object, u *map[string]interface{}) error {
+	err := toUnstructured(reflect.ValueOf(obj).Elem(), reflect.ValueOf(u).Elem())
+	if c.mismatchDetection {
+		newUnstr := &map[string]interface{}{}
+		newErr := toUnstructuredViaJSON(obj, newUnstr)
+		if (err != nil) != (newErr != nil) {
+			glog.Fatalf("ToUnstructured unexpected error: %v, expected: %v", err, newErr)
```

Comment:
I don't understand this comment- glog.Fatalf also gives the stacktrace.

I also don't understand the second part of the comment - if one is error and the other is not, objects doesn't make much sense (at least in case of those that returned error).